SRC/KEYB.H

```diff
@@ -1,52 +1,24 @@
 #ifndef __TK__KEYB__
 #define __TK__KEYB__
 
+#include <SDL.h>
 #include "PORT.H"
 
-struct KeyboardHandler
+namespace tk_port
 {
-    char state[256];
-
-    void init( int )
-    {
-        clear_stack();
-    }
-
-    void deinit()
-    {}
-
-    void clear_stack()
-    {
-        for ( int i = 0; i < 256; i++ )
-        {
-            state[i] = 0;
-        }
-        _kbhit_flag = 0;
-        _last_getch = 0;
-    }
-
-    int _kbhit_flag;
-    int _last_getch;
-
-    int kbhit()
-    {
-        if ( _kbhit_flag )
-        {
-            _kbhit_flag = 0;
-            return 1;
-        }
-        return 0;
-    }
-
-    int getch()
-    {
-        while (!kbhit())
-        {
-            tk_port::event_tick();
-        }
-        return _last_getch;
-    }
-
-};
+namespace keyb
+{
+    extern char state[256];
+
+    void init( int );
+    void deinit();
+    void clear_stack();
+    int kbhit();
+    int getch();
+    void handle_event( const SDL_Event * );
+}
+}
+
+namespace k = tk_port::keyb;
```

Comment:
In general it's not encouraged to declare namespace aliases in header files (as opposed to src files) because it is now visible to all files including this header. It increases the risk of name clash and is slightly confusing when one needs to track down where the `k` is defined as it behaves like a "global" thing.

But I don't consider it as a huge issue in this context, it allows more original feel in code.

Reply:
Yeah, this keeps the original feel, and `k` was a global variable in `GLOBVARS.H` already anyway :)